pytorch_lightning/trainer/trainer.py

```diff
@@ -1208,22 +1208,21 @@ def run_pretrain_routine(self, model: LightningModule):
         self.train()
 
     def _run_sanity_check(self, ref_model, model):
-
         using_val_step = ref_model.val_dataloader is not None and self.is_overridden('validation_step')
         should_sanity_check = using_val_step and self.num_sanity_val_steps > 0 and self.limit_val_batches > 0
 
         # run tiny validation (if validation defined)
         # to make sure program won't crash during val
         if should_sanity_check:
             self.reset_val_dataloader(ref_model)
+            self._num_sanity_val_steps = [min(self.num_sanity_val_steps, val_batches)
```

Comment:
since we have already fields for num_training_batches etc, should we call this simply num_sanity_batches for consistency? 
I guess the reason why you made it protected is because you want to differentiate between the user input arg and the internal list.

Reply:
yeah that is one reason, the other one is when using `lr_find` or `scale_batch_size` or any other case when `trainer.fit` is called more than once. In such cases I don't think changing the init parameters itself is a good idea, tests will fail too, that's why I used `_num_sanity_val_steps`.

Actually there are two ways to handle this:
1. create `_num_sanity_val_steps` once and use it again while initializing `progressbar.total` in the other PR.
2. or do this `num_batches = [min(self.num_sanity_val_steps, val_batches) for val_batches in self.num_val_batches]` again while calculating `progressbar.total` and avoid `_num_sanity_val_steps`.

Reply:
I prefer 1. since this is consistent with the other totals we compute internally. If possible, I would also prefer if we named this internal variable num_sanity_val_batches, also for consistency with the other totals. no strong preference though, the important part is that the implementation is clean fulfills our needs :)

Reply:
ping me if you need more help with this pr.

Reply:
@awaelchli yeah was thinking to change it to `num_sanity_val_batches`. I just clarifications for https://github.com/PyTorchLightning/pytorch-lightning/pull/2917#issuecomment-672994381.